### PR TITLE
Mobile compatibility

### PR DESCRIPTION
**Pull request by guerrap**
The `DatePicker`  component was causing weird horizontal scrolling on mobile devices since the picker contained by the popover is already on the DOM (with 0 opacity).
Having the `DatePicker` popover as a modal on mobile solves the issue.

### CODE REVIEW

File: packages/ui/src/components/input/date-time/index.tsx [hunk 1]
```diff
@@ -37,16 +44,36 @@ export const DateTimeInput = forwardRef<HTMLInputElement, DateTimeInputProps>(
         const [anchorEl, setAnchorEl] = useState<HTMLInputElement | null>(null);
         const popoverRef = useRef<HTMLDivElement>(null);
         const [open, setOpen] = useState(false);
+        const [modal, setModal] = useState(false);
+
+        const resizeObserver = useRef(
```
Comment (luzzif) [inline packages/ui/src/components/input/date-time/index.tsx:49]: Not sure a ref is necessary here. If we declare the `ResizeObserver` in the `useEffect` hook below directly it should be the same (minus the ref we wouldn't use) as it's not used anywhere else.

File: packages/ui/src/components/input/date-time/index.tsx [hunk 2]
```diff
@@ -37,16 +44,36 @@ export const DateTimeInput = forwardRef<HTMLInputElement, DateTimeInputProps>(
...
+            return () => {
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                resizeObserver.current.unobserve(document.body);
+            };
+        }, []);
+
         const handlePickerOpen = useCallback(() => {
             setOpen(true);
         }, []);
+        const handlePickerClose = useCallback(() => {
```
Comment (luzzif) [inline packages/ui/src/components/input/date-time/index.tsx:74]: Missing new line here

File: packages/ui/src/components/input/date/index.tsx [hunk 3]
```diff
@@ -37,16 +44,36 @@ export const DateInput = forwardRef<HTMLInputElement, DateInputProps>(
...
+            return () => {
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                resizeObserver.current.unobserve(document.body);
+            };
+        });
+
         const handlePickerOpen = useCallback(() => {
             setOpen(true);
         }, []);
+        const handlePickerClose = useCallback(() => {
```
Comment (luzzif) [inline packages/ui/src/components/input/date/index.tsx:74]: Ditto here

**Review by luzzif — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/input/date/index.tsx [hunk 4]
```diff
@@ -37,16 +44,36 @@ export const DateInput = forwardRef<HTMLInputElement, DateInputProps>(
...
             setOpen(false);
         });
 
+        useEffect(() => {
+            resizeObserver.current.observe(document.body);
+            return () => {
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                resizeObserver.current.unobserve(document.body);
+            };
+        });
```
Comment (Diogomartf) [inline packages/ui/src/components/input/date/index.tsx:69]: this useEffect is the same as above, as it is the `resizeObserver = useRef`

Maybe there is an opportunity to extract it.

Comment (guerrap) [reply packages/ui/src/components/input/date/index.tsx:69]: I'd leave it like this for now, since it's just a single `useEffect`.

**Review by luzzif — APPROVED** [review]

**Pull request merged**